src/components/Header/NavComponents/chain-display.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo } from "react";
+import {
+  Text,
+  useColorModeValue as mode,
+  Button,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  Stack,
+  PopoverArrow,
+  Center,
+} from "@chakra-ui/react";
+import { useChainAddresses } from "../../../utils/chainAddresses";
+import { changeChain } from "../../../utils/changeChain";
+import { internalAddressesPerNetworkId } from "../../../utils/contracts/contractAddresses/internalAddresses";
+import { UnsupportedChainIdError, useWeb3React } from "@web3-react/core";
+
+export const ChainSelector: React.FC<{}> = () => {
+  const chainAddresses = useChainAddresses();
+  const { error } = useWeb3React();
+
+  // Buttons to change to every available chain
+  const popoverData = useMemo(
+    () =>
+      Object.entries(internalAddressesPerNetworkId).map(([id, chain]) => {
+        return (
+          <Button
+            bg={mode({ base: "primary.500", md: "primary.500" }, "primary.500")}
+            colorScheme="teal"
+            size="xl"
+            h="40px"
+            onClick={() => changeChain(chain.chainName)}
+          >
```

Comment:
Lists generated dynamically should have keys in React so it can update them efficiently; use the ID from the parameters to help out by adding a key like this:
`key={id}`

Reply:
Did this myself in my commit, check it out and resolve this comment when you see it.